datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -836,7 +836,9 @@ fn add_roundrobin_on_top(
     n_target: usize,
 ) -> Result<DistributionContext> {
     // Adding repartition is helpful:
-    if input.plan.output_partitioning().partition_count() < n_target {
+    if input.plan.output_partitioning().partition_count() < n_target
+        || input.plan.children().is_empty()
+    {
```

Comment:
```suggestion
    if input.plan.output_partitioning().partition_count() < n_target
        || (input.plan.children().is_empty() && input.plan.output_partitioning().partition_count() > 1)
    {
```
I think this check handles the intended case better.

Reply:
Thanks for the suggestion, I agree (not repartitioning n=1 case)